src/System.Memory/src/System/Buffers/SequenceReader.cs

```diff
@@ -147,10 +147,13 @@ public bool TryRead(out T value)
         /// <summary>
         /// Move the reader back the specified number of items.
         /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown if trying to rewind a negative amount or more than <see cref="Consumed"/>.
+        /// </exception>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Rewind(long count)
         {
-            if (count < 0)
+            if (count < 0 || count > Consumed)
```

Comment:
This can be written more optimally as follows:
```C#
if ((ulong)count > (ulong)Consumed)
```

[sharplab.io](https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAGABJgRgG4BYAKEwGZ8AmXAYVwG9zc3dX2rcAbAewB2AcwaDkAV1QBTACbNcQqXCK4ADgCcAlgDdEcKbmRKVAX05tz+asD58euALIEAFP2G4Axn3EC4ASmZLdnwAdlxnLx84XAAeXDx0dE9vX1wAPlEBCWkZP1IyYLMC9ktuGztHGldBEUjfAJZi4LZMMOdxNyE/OuiM9s6/ejFJWXzC8hMgA==)

```suggestion
            if ((ulong)count > (ulong)Consumed)
```